src/out/ICFP18evaluation/evaluationCNN/Lantern/Lantern.cpp

```diff
@@ -46,7 +46,9 @@
       void *waterMark  = mallocBase;
       void* myMalloc(size_t bytes) {
         void* res = mallocAddr;
-        mallocAddr += bytes;
+        char* tmp = (char*) mallocAddr;
+        tmp += bytes;
+        mallocAddr = (void*) tmp;
         return res;
```

Comment:
I'd have a slight preference to write this in one line: `mallocAddr = (void*)((char*)mallocAddr + bytes);`

Reply:
Thanks! Done in https://github.com/feiwang3311/Lantern/pull/2/commits/91be36b8dfea6f98ea738badc42e44c5fcdde0cb.